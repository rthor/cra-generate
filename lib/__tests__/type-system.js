'use strict'

const typeSystem = require('../type-system')

const jsMock = `
function testFunction(arg) {
  return arg * 2
}

console.log(testFunction('foo'))
`.trim()

const flowMock = `
// @flow

function testFunction(arg) {
  return arg * 2
}

console.log(testFunction('foo'))
`.trim()

describe('typeSystem', () => {
  test('defaults to regular JS', () => {
    const actual = typeSystem(null, jsMock)
    expect(actual).toEqual({ content: jsMock })
  })

  test('supports flow', () => {
    const actual = typeSystem('flow', jsMock)
    expect(actual).toEqual({ content: flowMock })
  })
})
