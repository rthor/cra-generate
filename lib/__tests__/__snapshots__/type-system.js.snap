exports[`typeSystem supports flow 1`] = `
Object {
  "content": "// @flow


function testFunction(arg) {
  return arg * 2
}
",
}
`;
